classes/local/datasource/traits/hashcache.php

```diff
@@ -0,0 +1,70 @@
+<?php
+// This file is part of Moodle - https://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace mod_cms\local\datasource\traits;
+
+use mod_cms\local\lib;
+
+/**
+ * A trait to implement functions that make use of hash based cache keys.
+ *
+ * @package   mod_cms
+ * @author    Jason den Dulk <jasondendulk@catalyst-au.net>
+ * @copyright 2023, Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+trait hashcache {
+    /**
+     * Update the cache key fragment for the instance.
+     */
+    public function update_instance_cache_key() {
```

Comment:
Given that we are using the `get_instance_cache_key()` nomenclature, should we change this to `set_instance_cache_key()`?

Reply:
Using 'set' implies that we should be supplying the value to the system, when we are not.